### DISCUSSION TITLE
Unable to use navigateTo inside onResponseError inside useAsyncData or useFetch

### DISCUSSION THREAD

**Discussion by RomanSkrypnik** [nuxt/nuxt#22615] [Questions]
Calling `useFetch` in global middleware and using `onResponseError` interceptor to handle API error depending on its status code
While catching an error from API, I'm facing this error: 

`A composable that requires access to the Nuxt instance was called outside of a plugin, Nuxt hook, Nuxt middleware, or Vue setup function.`

It happens because middleware works on server side and I'm using `navigateTo` inside `onResponseError` which loses context
I have tried to use `callWithNuxt` method but unfortunately facing another error:

`TypeError: Cannot read properties of undefined (reading 'runWithContext')`

Is it possible to restore context inside this method?

**Comment by sync42johnny** [comment]
I've faced the same issue with navigateTo in onResponseError when using useFetch in Nuxt middleware(

**Comment by yuvalkarif** [comment]
I'm not sure if its on purpose that  `onResponseError` does not have its own nuxt context, but for the meanwhile I've managed to get it working with the [following repro](https://stackblitz.com/edit/nuxt-discussions-22615?file=app.vue,pages%2Ffoo.vue,pages%2Fbazz.vue).
Start by pressing pressing `Open in New Tab` because of url navigation, but go to `/foo` and you should end up in `/bazz`.
See the code in bazz to see how I've provided context to it.

**Reply by RomanSkrypnik** [reply]
Your code is executed in `setup`, it won't lose its context of course
My code is executed on server in middleware where `useFetch` is called - its `onResponse`, `onResponseError`, `onRequest` and `transform` methods lose nuxt context
It's quite a problem, because I have placed this interceptor callbacks in separate files and they are used not only in `useFetch` but also in `$fetch` plugin

So, the problem is that I lose context while using `onResponse` in middleware on server
